### ISSUE TITLE
Set keyboard binding based on chosen system

### ISSUE THREAD

**Issue by osamaq** [Microsoft/vscode#62891]
This is kind of related to this issue: https://github.com/Microsoft/vscode/issues/17550 But as you can see, I'm making a feature request.

Default operating system keybindings on different OS's force different VScode keybindings on each OS. That is understandable. However, some users would dislike inconsistencies with keybindings when working on multiple OS's. If the binding is already occupiued by the OS, then the OS action would take precedence over VScode.

I've already faced issues with VScode shortcuts on Ubuntu 18.04 such as "duplicate line below/above" which by default result in switching work spaces up and down. I resolved it by manually unbinding the default system keybinding.

Since problems already exist, I suggest a new option to choose the default keybinding set for VScode based on _e.g._ selected OS from a drop down menu. This would be more convenient in my opinion, and I would only have to unbind the system shortcuts for the VScode shortcuts that I commonly use.

**Comment by chrmarti** [comment]
We already have OS-specific default keybindings.

**Comment by osamaq** [comment]
Yes. I am asking for the ability to change the default keybinding set. [Here is a terrible mock up for what the option would look like on VSCode linux.](https://i.imgur.com/la9izfz.png)

It should give the ability to switch between the default keybindings:
https://code.visualstudio.com/shortcuts/keyboard-shortcuts-linux.pdf
https://code.visualstudio.com/shortcuts/keyboard-shortcuts-windows.pdf
https://code.visualstudio.com/shortcuts/keyboard-shortcuts-macos.pdf

**Comment by sandy081** [comment]
Assigning to @alexandrudima as he owns resolving the keybindings.

**Comment by alexandrudima** [comment]
> Since problems already exist, I suggest a new option to choose the default keybinding set for VScode based on e.g. selected OS from a drop down menu. This would be more convenient in my opinion, and I would only have to unbind the system shortcuts for the VScode shortcuts that I commonly use.

Most of the commands have the same keybindings. We have deviated precisely where there were well established OS keybindings (e.g. home is cmd+left on the mac) and those further produced a ripple of occupied keybindings, etc.

The solution is to open vs code on the mac, copy the default keybindings.json and paste them in your linux keybindings.json. This will work well when running linux inside a VM inside the mac.

But I don't think we will ever want to implement what you describe, simply because we have already implemented what you describe. You just don't like the way we have resolved OS conflicts. In other words, 95% of the commands use the same keybindings, they diverge where keybindings were occupied by the OS, where there were existing well established keybindings on the platform, or where conflicts were caused by the previous 2 reasons.
